pandas/io/formats/format.py

```diff
@@ -1891,27 +1893,24 @@ def _trim_zeros_float(
     Trims zeros, leaving just one before the decimal points if need be.
     """
     trimmed = str_floats
-    number_regex = re.compile(fr"\s*[\+-]?[0-9]+(\{decimal}[0-9]*)?")
+    number_regex = re.compile(fr"^\s*[\+-]?[0-9]+\{decimal}[0-9]*$")
 
-    def _is_number(x):
+    def is_number_with_decimal(x):
         return re.match(number_regex, x) is not None
 
-    def _cond(values):
-        finite = [x for x in values if _is_number(x)]
-        has_decimal = [decimal in x for x in finite]
-
-        return (
-            len(finite) > 0
-            and all(has_decimal)
-            and all(x.endswith("0") for x in finite)
-            and not (any(("e" in x) or ("E" in x) for x in finite))
-        )
+    def should_trim(values):
```

Comment:
can you type this (e.g. returns a bool)

and add a simple doc-string; was confused how you are using this, e.g. its working on *all* the numbers.